spec/lib/manageiq/api/common/paginated_response_spec.rb

```diff
@@ -0,0 +1,66 @@
+describe ManageIQ::API::Common::PaginatedResponse do
+  context "private links methods" do
+    let(:base_query) { double("AR:Clause", :count => count) }
+    let(:request) { double("Request", :original_url => "http://example.com/resource?param1=true&limit=#{limit}") }
+
+    def url_with_offset(offset)
+      "http://example.com/resource?limit=#{limit}&offset=#{offset}&param1=true"
+    end
+
+    context "with 6 records, limit 2" do
+      let(:count) { 6 }
+      let(:limit) { 2 }
```

Comment:
might be better to pick numbers that don't align mathematically (for example, count being 7, and limit being 2 would fine better edge cases).

Reply:
That's why I have (6 / 2) and (31 / 10), I wanted to test both cases.

Reply:
Ah ok, then as per my comment below, the context titles should probably be "when the count and limit align" and "when the count and limit do not align", which makes their purpose clearer and explains 2 tests.